src/Sulu/Bundle/AdminBundle/Resources/js/components/Select/Select.js

```diff
@@ -0,0 +1,105 @@
+// @flow
+import React from 'react';
+import {action, computed, observable} from 'mobx';
+import {observer} from 'mobx-react';
+import classnames from 'classnames';
+import selectStyles from './select.scss';
+import Divider from './Divider';
+import OverlayList from './OverlayList';
+import Option from './Option';
+import Label from './Label';
+
+@observer
+export default class Select extends React.PureComponent {
+    props: {
+        value?: string,
+        className: string,
+        onChange?: (s: string) => void,
+        children: Array<Option | Divider>,
+        icon?: string,
+    };
+
+    static defaultProps = {
+        children: [],
+        className: '',
+    };
+
+    label: Label;
+    @observable isOpen: boolean;
+
+    @action openList = () => {
+        this.isOpen = true;
+    };
+
+    @action closeList = () => {
+        this.isOpen = false;
+    };
+
+    @computed get labelText(): string {
+        let label = '';
+        React.Children.forEach(this.props.children, (child) => {
+            if (!label || this.props.value === child.props.value) {
```

Comment:
Is there a way to jump out the forEach callback? `.filter` is not possible because if `this.props.value` is not set `label` should be set to the content of the first child.